crates/python/src/executable.rs

```diff
@@ -108,7 +108,7 @@ impl PyExecutable {
         quil: String,
         registers: Vec<String>,
         parameters: Vec<PyParameter>,
-        #[pyo3(from_py_with = "crate::from_py::optional_non_zero_u16")] shots: Option<NonZeroU16>,
+        #[pyo3(from_py_with = "crate::from_py::optional_non_zero_u16")] shots: Option<NonZeroU32>,
```

Comment:
Linting failure on `crate::from_py::optional_non_zero_u16`. Should probably be this?

```suggestion
        #[pyo3(from_py_with = "crate::from_py::optional_non_zero_u32")] shots: Option<NonZeroU32>,
```